compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -167,12 +167,17 @@ object SymDenotations {
             println(i"${"  " * indent}completed $name in $owner")
           }
         }
-        else {
-          if (myFlags.is(Touched))
-            throw CyclicReference(this)(using ctx.withOwner(symbol))
-          myFlags |= Touched
-          atPhase(validFor.firstPhaseId)(completer.complete(this))
-        }
+        else
+          val traceCycles = CyclicReference.isTraced
+          try
+            if traceCycles then
+              CyclicReference.pushTrace("complete the info of ", symbol, "")
```

Comment:
`"complete the info of"` will be shown to users with `-explain-cyclic`. This terminology might not be clear to them. Maybe it should be something along the lines of `"computing the signature of"`